server/Assignments_filler/sheetFiller.py

```diff
@@ -0,0 +1,302 @@
+        else:
+            oldNID = contractList[0][0]
+            oldContractName = contractList[0][1]
+    else:
+        oldNID = contractList[0][0]
+        oldContractName = contractList[0][1]
+    query = (f"SELECT `created` FROM drupal.comment where nid={oldNID} ORDER BY created DESC")
+    oldCommentList = get_from_db(query)
+    oldNumberOfComments = len(oldCommentList)
+    if (oldNumberOfComments):
+        oldLastUpdated = oldCommentList[0][0]
+    else:
+        oldLastUpdated = 1
+    logging.info("Starting Checker")
+
+    while True:
+        #read new contract name from DB
+        writtenName = readSheet(service, readRange)
+        query = (f"SELECT `nid`,`title` FROM drupal.node where title like \"%{writtenName}%\" ORDER BY nid DESC")
+        contractList = get_from_db(query)
+        queryLength = len(contractList)
+        if (queryLength == 0):
+            if (not isRed):
+                paintSheet(service, "red")
+                isRed = True
+            time.sleep(refreshRate)
+            continue
+        elif (queryLength > 1):
+            for sublist in contractList:
+                exactName = sublist[1]
+                if (writtenName == exactName):
+                    exactIndex = contractList.index(sublist)
+                    newNID = contractList[exactIndex][0]
+                    newContractName = contractList[exactIndex][1]
+                    nameLog = (f"found exact match among {queryLength} contracts that contain \"{writtenName}\"")
+                    break
+            else:
+                newNID = contractList[0][0]
+                newContractName = contractList[0][1]
+                nameLog = (f"could not find exact match among {queryLength} contracts that contain \"{writtenName}\", choosing the latest...")
+        else:
+            newNID = contractList[0][0]
+            newContractName = contractList[0][1]
+            nameLog = "found single contract and chose it!"
+
+        #check last updated comment and number of comments
+        query = (f"SELECT `created` FROM drupal.comment where nid={newNID} ORDER BY created DESC")
+        commentList = get_from_db(query)
+        newNumberOfComments = len(commentList)
+        if (newNumberOfComments):
+            newLastUpdated = commentList[0][0]
+        else:
+            newLastUpdated = 1
+        
+        #compare the new and the old data
+        if (newContractName != oldContractName):
+            logging.info(f"New contract name detected! {nameLog}")
+            nameChanged = True
+            break
+        elif (newNumberOfComments != oldNumberOfComments):
+            logging.info("A comment was added!")
+            nameChanged = False
+            break
+        elif (newLastUpdated != oldLastUpdated):
+            logging.info("A comment was modified!")
+            nameChanged = False
+            break
+        time.sleep(refreshRate)
+    filler(service, nameChanged, isRed, newContractName, newNID)
```

Comment:
You should define a default value for `nameChanged`, it might not be set as well. Won't error because `bool(None)` is still `False`, but good practice.